kubechain/internal/controller/taskrun/taskrun_controller.go

```diff
@@ -438,9 +440,14 @@ func (r *TaskRunReconciler) processLLMResponse(ctx context.Context, output *kube
 		// End the parent span since we've reached a terminal state
 		r.endTaskRunSpan(ctx, taskRun, codes.Ok, "TaskRun completed successfully with final answer")
 	} else {
+		// Generate a unique ID for this set of tool calls
+		toolCallRequestId := uuid.New().String()[:7] // Using first 7 characters for brevity
+		logger.Info("Generated toolCallRequestId for tool calls", "id", toolCallRequestId)
+
 		// tool call branch: create TaskRunToolCall objects for each tool call returned by the LLM.
 		statusUpdate.Status.Output = ""
 		statusUpdate.Status.Phase = kubechainv1alpha1.TaskRunPhaseToolCallsPending
+		statusUpdate.Status.ToolCallRequestId = toolCallRequestId
```

Comment:
so...this is a mutable field that tracks whatever "set of tool calls" we're waiting on - while this seems to work, I'm 🤔 about other design options